src/app/static/src/app/components/plots/table/Table.vue

```diff
@@ -91,24 +91,27 @@ export default Vue.extend<{}, {}, Computed, Props>({
              return this.filters.filter((f: Filter) => f.id != this.areaFilterId);
         },
         areaFilter() {
-          if (this.selections.detail === 0 || !this.selections.detail) {
-            return this.countryAreaFilterOption.find((f: Filter) => f.id == this.areaFilterId)!!;
-          } else return this.filters.find((f: Filter) => f.id == this.areaFilterId)!!;
+          return this.filters.find((f: Filter) => f.id == this.areaFilterId)!!;
         },
         flattenedAreas() {
-                 return this.areaFilter ? flattenOptions(this.areaFilter.options) : {};
+          if (this.selections.detail === 0 || !this.selections.detail){
+            return this.areaFilter ? flattenOptions([this.countryAreaFilterOption]) : {};
+          }
+          return this.areaFilter ? flattenOptions(this.areaFilter.options) : {};
         },
         selectedAreaIds() {
             const selectedAreaIdSet = flattenToIdSet(this.selectedAreaFilterOptions.map(o => o.id), this.flattenedAreas);
             const areaArray = Array.from(selectedAreaIdSet)
             if (this.selections.detail === 0 || !this.selections.detail){
-                return [areaArray[0]]
+                return areaArray[0].length === 3 ? [areaArray[0]] : []
```

Comment:
This seems a bit suspect, why `===3`? Because that's the number of regions at level 1? What if it was dealing with a country with 4 top level regions? 

I think this can be simplified further. Really, we just want selectedAreaFilterOptions to always give back the right options, and then selectedAreaIds can just map those to the right Ids (allowing for country level ids having different format). 

How about changing this to:
```
 selectedAreaIds() {
            const selectedAreaIdSet = flattenToIdSet(this.selectedAreaFilterOptions.map(o => o.id), this.flattenedAreas);
            const areaArray = Array.from(selectedAreaIdSet);
            if (this.selections.detail === 0){
                return [areaArray[0]]
            } else return areaArray.filter(val => parseInt(val[4]) === this.selections.detail);
        },
        selectedAreaFilterOptions() {
            const selectedOptions = this.selections.selectedFilterOptions[this.areaFilterId];
            if(this.selections.detail === 0){
                return this.countryAreaFilterOption ? [this.countryAreaFilterOption] : []
            }
            else if (selectedOptions && selectedOptions.length > 0) {
                return selectedOptions
            } else return this.areaFilter ? this.areaFilter.options : []; //consider all top level areas to be selected if none are
        },
```

Reply:
> This seems a bit suspect, why `===3`? Because that's the number of regions at level 1? What if it was dealing with a country with 4 top level regions?
> 
> I think this can be simplified further. Really, we just want selectedAreaFilterOptions to always give back the right options, and then selectedAreaIds can just map those to the right Ids (allowing for country level ids having different format).
> 
> How about changing this to:
> 
> ```
>  selectedAreaIds() {
>             const selectedAreaIdSet = flattenToIdSet(this.selectedAreaFilterOptions.map(o => o.id), this.flattenedAreas);
>             const areaArray = Array.from(selectedAreaIdSet);
>             if (this.selections.detail === 0){
>                 return [areaArray[0]]
>             } else return areaArray.filter(val => parseInt(val[4]) === this.selections.detail);
>         },
>         selectedAreaFilterOptions() {
>             const selectedOptions = this.selections.selectedFilterOptions[this.areaFilterId];
>             if(this.selections.detail === 0){
>                 return this.countryAreaFilterOption ? [this.countryAreaFilterOption] : []
>             }
>             else if (selectedOptions && selectedOptions.length > 0) {
>                 return selectedOptions
>             } else return this.areaFilter ? this.areaFilter.options : []; //consider all top level areas to be selected if none are
>         },
> ```

The idea of the ```===3``` was to prevent the table simply showing the first value in the area array when detail === 0 and some of the sub areas have been selected. For example, if Northern and Central were selected but detail was set to 0, the table would return the Northern row (the first value in the array). The country level area id string is always 3 characters long (I believe), eg, Malawi is MWI. So for the above example, the code I wrote return no data available are for these selections instead, which is more accurate.

Reply:
Oh I see - ok, but it's probably even simpler just to use the countryLevelFilterOption you get back from selectedAreFilterOption if detail === 0 - that's what it's there for.

Reply:
Actually simplest of all would be to combine the two methods since nothing else uses them and since they're both needing to distinguish detail == 0 from other cases it's probably less confusing to combine them. But that can be a refactor for another day.

Reply:
I'll merge this in. This will have the effect of always returning the country level row if detail is set 0, no matter what is selected in the area filter, rather than no data are available for these selections.